### PR TITLE
[CARBONDATA-1836][Spark-2.2] Carbon Spark2.2 Integration Resolve CatalogRelation

### PR DESCRIPTION
**Pull request by sounakr**
Code change in order to support change in CatalogRelation case class name change to HiveTableRelation. 

 - [ ] Any interfaces changed?
      No
 
 - [ ] Any backward compatibility impacted?
      No
 
 - [ ] Document update required?
      No

 - [ ] Testing done
      Yes
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
   NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1567/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1966/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1599/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1995/

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala [hunk 1]
```diff
@@ -147,7 +147,11 @@ class CarbonFileMetastore extends CarbonMetaStore {
       case LogicalRelation(
       carbonDatasourceHadoopRelation: CarbonDatasourceHadoopRelation, _, _) =>
         carbonDatasourceHadoopRelation.carbonRelation
-      case SubqueryAlias(_, c: CatalogRelation) if sparkSession.version.startsWith("2.2") =>
+      case SubqueryAlias(_, c) if sparkSession.version.startsWith("2.2") &&
+      (c.getClass.getName.equals("org.apache.spark.sql.catalyst.catalog.CatalogRelation") ||
+       c.getClass.getName.equals("org.apache.spark.sql.catalyst.catalog.HiveTableRelation") ||
+       c.getClass.getName
+         .equals("org.apache.spark.sql.catalyst.catalog.UnresolvedCatalogRelation")) =>
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala]: Please format as below
```
c.getClass.getName.equals(
  "org.apache.spark.sql.catalyst.catalog.UnresolvedCatalogRelation"))
```

Comment (sounakr) [reply integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1638/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2028/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/590/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/592/

**Comment by ravipesala** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2190/

**Pull request closed without merging**